src/effects/VST/VSTEffect.h

```diff
@@ -199,6 +204,33 @@ struct VSTEffectWrapper : public VSTEffectLink, public XMLTagHandler
    void SaveFXP(const wxFileName& fn) const;
    void SaveFXProgram(wxMemoryBuffer& buf, int index) const;
 
+   // VST plugin -> host callback
+   static intptr_t AudioMaster(AEffect *effect,
+                               int32_t opcode,
+                               int32_t index,
+                               intptr_t value,
+                               void * ptr,
+                               float opt);
+
+   // These are needed to move the AudioMaster callback from VSTEffect to this struct
+   //
+   static intptr_t mCurrentEffectID;
+   virtual void NeedIdle() = 0;
+   virtual void UpdateDisplay() = 0;
+   virtual VstTimeInfo* GetTimeInfo() = 0;
+   virtual void SetBufferDelay(int samples) = 0;
+   virtual float GetSampleRate() = 0;
+   virtual int GetProcessLevel() = 0;
+   virtual void SizeWindow(int w, int h) = 0;
+   virtual void Automate(int index, float value) = 0;
+
+   // Loaded once when getting common VST things;
+   // placed here becaue they will be reused when an instance is created
+   //
+   vstPluginMain mPluginMain;   // entry point in the plugin dll
```

Comment:
This is just a "C" pointer to function -- so initialize it to null

Reply:
ok